apps/server/src/modules/provisioning/strategy/oidc/service/oidc-provisioning.service.spec.ts

```diff
@@ -101,102 +101,193 @@ describe('OidcProvisioningService', () => {
+				});
+			});
+
+			describe('when the external system does not provide a location for the school', () => {
+				it('should only use the school name', async () => {
+					const { systemId, externalSchoolDto, savedSchoolDO } = setup();
+					externalSchoolDto.location = undefined;
+					savedSchoolDO.name = 'name';
+
+					await service.provisionExternalSchool(externalSchoolDto, systemId);
+
+					expect(schoolService.save).toHaveBeenCalledWith({ ...savedSchoolDO, id: undefined }, true);
+				});
+			});
 		});
 
 		describe('when external school already exist', () => {
-			it('should update the existing school', async () => {
-				const { systemId, externalSchoolDto, existingSchoolDO, savedSchoolDO } = setup();
+			const setup = () => {
+				const systemId = 'systemId';
+				const externalSchoolDto: ExternalSchoolDto = new ExternalSchoolDto({
+					externalId: 'externalId',
+					name: 'name',
+					officialSchoolNumber: 'officialSchoolNumber',
+				});
+
+				const schoolYear = schoolYearFactory.build();
+				const federalState = federalStateFactory.build();
+				const savedSchoolDO = legacySchoolDoFactory.build({
+					id: 'schoolId',
+					externalId: 'externalId',
+					name: 'name',
+					officialSchoolNumber: 'officialSchoolNumber',
+					systems: [systemId],
+					features: [SchoolFeatures.OAUTH_PROVISIONING_ENABLED],
+					schoolYear,
+					federalState,
+				});
+				const existingSchoolDO = legacySchoolDoFactory.build({
+					id: 'schoolId',
+					externalId: 'externalId',
+					name: 'existingName',
+					officialSchoolNumber: 'existingOfficialSchoolNumber',
+					systems: [systemId],
+					features: [SchoolFeatures.OAUTH_PROVISIONING_ENABLED],
+				});
 
+				schoolService.save.mockResolvedValue(savedSchoolDO);
 				schoolService.getSchoolByExternalId.mockResolvedValue(existingSchoolDO);
+				schoolYearService.getCurrentSchoolYear.mockResolvedValue(schoolYear);
+				federalStateService.findFederalStateByName.mockResolvedValue(federalState);
+
+				return {
+					systemId,
+					externalSchoolDto,
+					savedSchoolDO,
+					existingSchoolDO,
+				};
+			};
+
+			it('should update the existing school', async () => {
+				const { systemId, externalSchoolDto, savedSchoolDO } = setup();
 
 				const result: LegacySchoolDo = await service.provisionExternalSchool(externalSchoolDto, systemId);
 
 				expect(result).toEqual(savedSchoolDO);
 			});
 
-			it('should append the new system', async () => {
-				const { systemId, externalSchoolDto, existingSchoolDO, savedSchoolDO } = setup();
-				const otherSystemId = 'otherSystemId';
-				existingSchoolDO.systems = [otherSystemId];
-
-				schoolService.getSchoolByExternalId.mockResolvedValue(existingSchoolDO);
+			describe('when the external system provides a location for the school', () => {
+				it('should append it to the school name', async () => {
+					const { systemId, externalSchoolDto, savedSchoolDO } = setup();
```

Comment:
I'd prefer to use a own setup and create the schooldata in a separate function. We have to split it up in the end anyway